addons/sourcemod/scripting/BoostAlert.sp

```diff
@@ -2,122 +2,276 @@
+			iMinDamageByWeapon = 120;
+		// Pistols
+		if (strcmp(sWeapon, "deagle", false) == 0 || strcmp(sWeapon, "revolver", false) == 0)
+			iMinDamageByWeapon = 80;
+		// Knife
+		if (iKnife != -1)
+			iMinDamageByWeapon = 1;
+
+		// Allow knife
+		if (iDamage < iMinDamageByWeapon || iMinDamageByWeapon == 0)
+			return Plugin_Continue;
+
+		int time = GetTime();
+		int iVictimTeam = GetClientTeam(victim);
+		int iAttackerTeam = GetClientTeam(attacker);
+		int diffTime = time - 1; // Prevent spam from ShotsGun (Firing multiple bullets at once)
+
+		g_iGameTimes[victim] = time;
+		g_iOriginalAttacker[victim] = attacker;
+		g_iOriginalAttackerID[victim] = GetClientUserId(attacker);
+		g_iNotificationTime[victim] = (time + g_CVar_BoostDelay.IntValue);
+
+		if (iKnife != -1)
 		{
-			int time = GetTime();
-			int victim = GetClientOfUserId(GetEventInt(hEvent, "userid"));
-			int attacker = GetClientOfUserId(GetEventInt(hEvent, "attacker"));
+			sType = "Knifed";
+			FormatEx(sDetails, sizeof(sDetails), "(-%d HP)", iDamage);
+		}
+		else
+		{
+			sType = "Boosted";
+			FormatEx(sDetails, sizeof(sDetails), "(-%d HP with %s)", iDamage, sWeapon);
+		}
 
-			g_iGameTimes[victim] = time;
-			g_iDamagedIDs[victim] = victim;
-			g_iAttackerIDs[victim] = attacker;
+		if ((iDamage > 35 || g_iGameTimeSpam[victim] <= diffTime) && iAttackerTeam == 3 && iVictimTeam == 2)
+		{
+			if (g_CVar_Logs.BoolValue)
+				LogMessage("%L %s %L %s", attacker, sType, victim, sDetails);
 
-			int diffTime = time - g_CVar_BoostSpam.IntValue;
+			FormatEx(sMessage, sizeof(sMessage), "{blue}%N {default}%s {red}%N{default}. %s", attacker, sType, victim, sDetails);
+			PrintMessage(sMessage);
 
-			if (g_iGameTimeSpam[victim] <= diffTime)
+			g_iGameTimeSpam[victim] = time;
+
+			Forward_OnAlert(attacker, victim, iDamage, sWeapon);
+		}
+		else if (iAttackerTeam == 2 && iVictimTeam == 3 && (g_iNotificationTime[attacker] >= time || iDamage > 35))
+		{
+			int iInitialAttacker = GetClientOfUserId(g_iOriginalAttackerID[attacker]);
+			if (victim != iInitialAttacker)
 			{
-				g_iGameTimeSpam[victim] = time;
-				for (int i = 1; i <= MaxClients; i++)
+				char sPluginZR[32], sOriginalAttacker[32];
+				sPluginZR = g_Plugin_ZR ? "infected" : "killed";
+				GetClientAuthId(iInitialAttacker, AuthId_Steam2, sOriginalAttacker, sizeof(sOriginalAttacker), false);
+
+				CPrintToChatAll("%s {red}%N {green}({lightgreen}%s{green}) {default}%s {blue}%N{default}.", CHAT_PREFIX, attacker, sAtkSID, sPluginZR, victim);
+
+				if (iInitialAttacker != 1)
+				{
+					if (g_CVar_Logs.BoolValue)
+						LogMessage("%L %s %L (Recently %s %s by %L)", attacker, sPluginZR, victim, sType, sDetails, iInitialAttacker);
+
+					CPrintToChatAll("%s {red}%N {default}was recently %s %s by{blue} %N{default}.", CHAT_PREFIX, attacker, sType, sDetails, iInitialAttacker);
+					Forward_OnKill(attacker, sAtkSID, victim, sVictSID, iInitialAttacker, sOriginalAttacker, iDamage, sWeapon);
+				}
+				else

```

Comment:
almost the same function as in the previous if, keep it dry according to this comment:
- https://github.com/srcdslab/sm-plugin-BoostAlert/pull/7/files#r1516088644